### PR TITLE
Add WindmillConnectionsCache used to consume GetWorkerMetadata response and updates

### PR DESCRIPTION
**Pull request by m-trieu**
WindmillConnectionsCache will be used to consume GetWorkerMetadata response and updates.

It exposes a separate read and write interface.  The write interface will be used for the caller of GetWorkerMetadata stream to be able to pass the response along to the cache to consume.  The read interface will be vended out to GetWorkStream, GetDataStream, and CommitWorkStream call sites in order to fetch connections directly to Streaming Engine Windmill Workers, or the Dispatcher.

WindmillConnectionCacheTokens will be used to attach to WorkItems to correctly call GetDataStream and CommitWorkStream.  

Since the WindmillConnectionsCache's will be shared, it uses AtomicReference and synchronization around writes to the cache.  Updates are all or nothing and done under a lock/synchronization, and reads fetch a snapshot of the of the AtomicReference's value to serve reads without locking.

r: @scwhittle 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/connectionscache/VendoredCredentialsAdapter.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import org.apache.beam.vendor.grpc.v1p54p0.com.google.auth.Credentials;
+import org.apache.beam.vendor.grpc.v1p54p0.com.google.auth.RequestMetadataCallback;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions;
+
+/**
+ * Create a wrapper around credentials that delegates to the underlying {@link
+ * com.google.auth.Credentials}. Note that this class should override every method that is not final
+ * and not static and call the delegate directly.
+ *
+ * <p>TODO: Replace this with an auto generated proxy which calls the underlying implementation
```
Comment (m-trieu) [inline runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/connectionscache/VendoredCredentialsAdapter.java:35]: @scwhittle do you have context on this (or knows who does)? I can just replace this class with whatever this comment is referring to, or add more text describing what needs to be done, but not quite sure what it is.

**Review by scwhittle — CHANGES_REQUESTED** [review]
Please ignore the nits for now, we can come back to them if still present with possible interface changes.

**Comment by damccorm** [comment]
@m-trieu what are next steps here?

**Comment by scwhittle** [comment]
I belive this is replaced by the other PRs, closing this one. Repush after rebasing if there are things worht keeping in this one.

**Pull request closed without merging**